feat(install): Support for addon resources in syndesis-operator

astefanutti opened the pull request:
Fixes #4381.

We may want to either:
* Add an option to deactivate add-on by default
* Or do not fail-fast on add-on resource creation error

heiko-braun commented:
@rhuss Can you take a look at this?

rhuss commented:
@astefanutti is this based on top of #4460 or on the state on master ?

astefanutti commented:
@rhuss it's based on `master` branch. Once #4460 is merged I can rebase this PR.

rhuss commented:
@astefanutti ok. I expect to be in a mergeable state tomorrow evening with #4460 I will review then this PR after your rebase.

astefanutti commented:
@rhuss thanks for the update. Sounds good to me!

astefanutti commented:
I've rebased the PR on top of #4460 and changed the original behaviour so that the `--addons` option must be explicitly set for the addons to be applied.

rhuss commented on install/operator/pkg/controller/syndesis/syndesis_controller.go:
```diff
@@ -39,12 +42,24 @@ func init() {
+		return nil, err
+	}
+	return &ReconcileSyndesis{
+		apis:   clientset,
```
Why are these apis needed ?

rhuss commented:
Looks good but conflicts a bit with the refactorings of #4574 where the client and scheme are put into a baseAction struct.

tbh didn't get it what this `apis` field is for. Is a scheme not good enough ?

astefanutti commented:
The `apis` field stores a references to the Kubernetes interface, under the `apis` endpoint, that gives access to the discovery API. It's used in the upgrade action to get the supported types to then delete the addons resources. While it is possible to create that interface client for each upgrade, it can be useful to initialise it a start time so that it's available in addition to the operator SDK client (with cache, informers, limited endpoints...).

rhuss commented:
sounds good. let's merge it.
